libs/parallelism/execution/include/hpx/execution/executors/rebind_executor.hpp

```diff
@@ -49,23 +49,42 @@ namespace hpx { namespace parallel { namespace execution {
 
     /// Rebind the type of executor used by an execution policy. The execution
     /// category of Executor shall not be weaker than that of ExecutionPolicy.
-    template <typename ExecutionPolicy, typename Executor, typename Parameters>
+    template <typename ExPolicy, typename Executor, typename Parameters>
     struct rebind_executor
     {
         /// \cond NOINTERNAL
-        typedef typename std::decay<Executor>::type executor_type;
-        typedef typename std::decay<Parameters>::type parameters_type;
+        using policy_type = std::decay_t<ExPolicy>;
+        using executor_type = std::decay_t<Executor>;
+        using parameters_type = std::decay_t<Parameters>;
 
-        typedef typename ExecutionPolicy::execution_category category1;
-        typedef typename hpx::traits::executor_execution_category<
-            executor_type>::type category2;
+        using category1 = typename policy_type::execution_category;
+        using category2 = typename hpx::traits::executor_execution_category<
+            executor_type>::type;
 
         static_assert(detail::is_not_weaker<category2, category1>::value,
             "detail::is_not_weaker<category2, category1>::value");
         /// \endcond
 
         /// The type of the rebound execution policy
-        typedef typename ExecutionPolicy::template rebind<executor_type,
-            parameters_type>::type type;
+        using type = typename policy_type::template rebind<executor_type,
+            parameters_type>::type;
     };
+
+    //////////////////////////////////////////////////////////////////////////
+    struct create_rebound_policy_t
+    {
+        template <typename ExPolicy, typename Executor, typename Parameters>
+        constexpr decltype(auto) operator()(
+            ExPolicy&&, Executor&& exec, Parameters&& parameters) const
+        {
+            using rebound_type =
+                typename rebind_executor<ExPolicy, Executor, Parameters>::type;
+
+            return rebound_type(std::forward<Executor>(exec),
+                std::forward<Parameters>(parameters));
+        }
+    };
+
+    HPX_INLINE_CONSTEXPR_VARIABLE create_rebound_policy_t
+        create_rebound_policy{};
```

Comment:
I have to ask only because this looks like it could be a plain function template: is this an object instead of a function template to make the friend declarations shorter?

Reply:
> I have to ask only because this looks like it could be a plain function template: is this an object instead of a function template to make the friend declarations shorter?

Yes.